sycl/include/CL/sycl/INTEL/online_compiler.hpp

```diff
@@ -0,0 +1,192 @@
+//===------- online_compiler.hpp - Online source compilation service ------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl/detail/defines_elementary.hpp> // for __SYCL_INLINE_NAMESPACE
+#include <CL/sycl/context.hpp>
+#include <CL/sycl/device.hpp>
+
+#include <memory>
+#include <vector>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace INTEL {
+
+using byte = unsigned char;
+
+enum class compiled_code_format {
+  SPIRV // the only format supported for now
+};
+
+class device_arch {
+public:
+  static constexpr int any = 0;
+
+  device_arch(int Val) : Val(Val) {}
+
+  enum gpu {
+    gpu_any = 1,
+    gpu_gen9,
+    gpu_skl = gpu_gen9,
+    gpu_gen9_5,
+    gpu_kbl = gpu_gen9_5,
+    gpu_cfl = gpu_gen9_5,
+    gpu_gen11,
+    gpu_icl = gpu_gen11,
+    gpu_gen12
+  };
+
+  enum cpu {
+    cpu_any = 1,
+  };
+
+  enum fpga {
+    fpga_any = 1,
+  };
+
+  operator int() { return Val; }
+
+private:
+  int Val;
+};
+
+/// Represents an error happend during online compilation.
+class online_compile_error : public sycl::exception {
+  // TBD
```

Comment:
It would be very convenient to have a constructor accepting 'const string_class &Msg'
```suggestion
  // TBD
  online_compiler_error() = default;
  online_compiler_error(const string_class &Msg);
```